pandas/src/sparse.pyx

```diff
@@ -191,6 +196,35 @@ cdef class IntIndex(SparseIndex):
         else:
             return -1
 
+    @cython.wraparound(False)
+    cpdef lookup_array(self, ndarray[int32_t, ndim=1] indexer):
+        cdef:
+            Py_ssize_t n, i, ind_val
+            ndarray[int32_t, ndim=1] inds
+            ndarray[int32_t, ndim=1] results
+
+        n = len(indexer)
+        results = np.empty(n, dtype=np.int32)
+
+        if self.npoints == 0:
+            results.fill(-1)
+            return results
+
+        inds = self.indices
+        for 0 <= i < n:
+            ind_val = indexer[i]
+            if ind_val < 0 or self.length <= ind_val:
+                results[i] = -1
+            else:
```

Comment:
I think you need to filter first, then call:

``inds.searchsorted(indices_to_find)`` IOW it takes an array as an input (and outputs and array).